compiler/src/linking/link.re

```diff
@@ -301,17 +285,25 @@ let rec globalize_names = (~function_names, ~global_names, ~label_names, expr) =
 };
 
 let table_offset = ref(0);
-let module_id = ref(0);
+let module_id = ref(Comp_utils.encoded_int32(0));
```

Comment:
This was a really sad bug that took away a week of my life.

Reply:
Can you explain it a bit?

Reply:
As defined in the data representations doc, module IDs are always tagged and thus GC-able. Except during linking, the linker assigned non-tagged IDS 🤦‍♂️

This wasn't a problem before because the IDs were treated by the runtime as untagged, but some updates while working on this PR caused them to be treated as regular values and made the bug show up.